app/controllers/test_run_controller.rb

```diff
@@ -8,17 +8,42 @@ def program_test_run_select_list
   end
 
   def show
-    @test_run = TestRun.find(params[:id])
-    render json: @test_run.test_run_info
+    test_run = TestRun.find(params[:id])
```

Comment:
TestRun.find is only in 3 of the 7 controller actions, so I would have to specify the ones that I am skipping (or not skipping), and create an instance variable for the found TestRun so those 3 actions have access to it.  Doing this would DRY up the controller at the expense of clarity.
http://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html
I learned some stuff!